### PR TITLE
Made reddit provisioning repeatable

### PR DESCRIPTION
**Pull request by rhysyngsun**
Sister PR to mitodl/open-discussions#299

This PR fixes the oauth client we use for local testing to a fixed value as well as make the provisioning very repeatable.

Recommend you checkout into a new directory and `vagrant halt` your other instance.

To test: `vagrant up` this branch and verify you can hit the reddit.local site. Sign in with `localdev`/`password1` and verify you see an app with the static client_id and secret.

`vagrant reload && vagrant provision` to be sure the site is still functional after a VM reload and provision.

Test in conjunction with the sister PR to verify that everything works together.

### CODE REVIEW

**Comment by noisecapella** [comment]
I'm seeing this error when running `vagrant provision` on a new VM:

    ==> default: stdin: is not a tty
    ==> default: Warning: g.media_domain == g.domain. This may give untrusted content access to user cookies
    ==> default: Warning: g.oauth_domain == g.domain. CORS requests to g.domain will be allowed
    ==> default: Overriding g.--plugin to r2
    ==> default: reddit:18192 started 10e2d3f at 19:08:36 (took 0.14s)
    ==> default: Traceback (most recent call last):
    ==> default:   File "/usr/bin/paster", line 4, in <module>
    ==> default:     command.run()
    ==> default:   File "/usr/lib/python2.7/dist-packages/paste/script/command.py", line 104, in run
    ==> default:     invoke(command, command_name, options, args[1:])
    ==> default:   File "/usr/lib/python2.7/dist-packages/paste/script/command.py", line 143, in invoke
    ==> default:     exit_code = runner.run(args)
    ==> default:   File "/usr/lib/python2.7/dist-packages/paste/script/command.py", line 238, in run
    ==> default:     result = self.command()
    ==> default:   File "/home/vagrant/src/reddit/r2/r2/commands.py", line 93, in command
    ==> default:     
    ==> default: execfile(self.args[1], loaded_namespace)
    ==> default:   File "/tmp/create_api_user.sh", line 14, in <module>
    ==> default:     
    ==> default: client = OAuth2Client._byID('od_client_id')
    ==> default:   File "/home/vagrant/src/reddit/r2/r2/lib/db/tdb_cassandra.py", line 382, in _byID
    ==> default:     
    ==> default: ids[0]))
    ==> default: r2.lib.db.tdb_cassandra
    ==> default: .
    ==> default: NotFound
    ==> default: : 
    ==> default: NotFound(
    ==> default:     _message="<OAuth2Client 'od_client_id'>",
    ==> default: message="<OAuth2Client 'od_client_id'>")
    The SSH command responded with a non-zero exit status. Vagrant
    assumes that this means the command failed. The output for this command
    should be in the log above. Please read the output to determine what
    went wrong.

**Comment by noisecapella** [comment]
Here is `tmp/create_api_user.sh`:

    #!/usr/bin/env python
    # vi: set ft=python :
    
    from r2.models import Account, register, OAuth2Client, change_password
    from r2.lib.db.thing import NotFound
    
    try:
        localdev = Account._by_name('localdev')
        change_password(localdev, 'password1')
    except NotFound:
        localdev = register('localdev', 'password1', '127.0.0.1')
    
    try:
        client = OAuth2Client._byID('od_client_id')
    except NotFound:
        client = OAuth2Client(_id='od_client_id',
                              secret='od_client_secret',
                              redirect_uri='https://localhost:8063/',
                              name='Client app for open discussions',
                              app_type='script')
    
    client.add_developer(localdev)
    client._commit()

File: scripts/create_api_user.sh [hunk 1]
```diff
@@ -0,0 +1,23 @@
+#!/usr/bin/env python
```
Comment (noisecapella) [inline scripts/create_api_user.sh]: Should this file have a `.py` extension instead?

**Review by noisecapella — CHANGES_REQUESTED** [review]

**Comment by rhysyngsun** [comment]
@noisecapella fixed

**Review by noisecapella — APPROVED** [review]

**Pull request merged**